src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeImageSelect.ts

```diff
@@ -0,0 +1,76 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { SingleValue } from 'react-select';
+import { ComputeImage } from 'src/analysis/useComputeImages';
+import { GroupedSelect } from 'src/components/common';
+
+export interface GcpComputeImageSelectProps {
+  readonly selectedComputeImageUrl: string;
+  readonly setSelectedComputeImageUrl: (value: string) => void;
+  readonly images: ComputeImage[];
+  readonly hasCustomOption: boolean;
+  readonly customOptionUrl: string;
+}
+
+interface ImageSelectOption {
+  value: string;
+  label?: string;
+}
+
+interface ImageSelectOptionGroup {
+  label: string;
+  options: ImageSelectOption[];
+}
+
+export const GcpComputeImageSelect: React.FC<GcpComputeImageSelectProps> = (props: GcpComputeImageSelectProps) => {
+  const {
+    selectedComputeImageUrl,
+    setSelectedComputeImageUrl,
+    images,
+    hasCustomOption,
+    customOptionUrl,
+    ...restProps
+  } = props;
+
+  const filterImages: (predicate: any) => ImageSelectOption[] = (predicate) =>
```

Comment:
```suggestion
  const filterImages = (predicate: any): ImageSelectOption[] =>
```

(In my opinion) this would be easier to read with types specified in the function definition instead of on the variable.